omnigibson/robots/behaviorbot.py

```diff
@@ -0,0 +1,677 @@
+FINGER_MID_LINK_NAMES = ("Tproximal", "Iproximal", "Mproximal", "Rproximal", "Pproximal")
+FINGER_TIP_LINK_NAMES = ("Tmiddle", "Imiddle", "Mmiddle", "Rmiddle", "Pmiddle")
+THUMB_LINK_NAME = "Tmiddle"
+
+
+class Behaviorbot(ManipulationRobot, LocomotionRobot, ActiveCameraRobot):
+    """
+    A humanoid robot that can be used in VR as an avatar. It has two hands, a body and a head with two cameras.
+    """
+
+    def __init__(
+            self,
+            # Shared kwargs in hierarchy
+            name,
+            prim_path=None,
+            class_id=None,
+            uuid=None,
+            scale=None,
+            visible=True,
+            fixed_base=True,
+            visual_only=False,
+            self_collisions=True,
+            load_config=None,
+
+            # Unique to USDObject hierarchy
+            abilities=None,
+
+            # Unique to ControllableObject hierarchy
+            control_freq=None,
+            controller_config=None,
+            action_type="continuous",
+            action_normalize=False,
+            reset_joint_pos=None,
+
+            # Unique to BaseRobot
+            obs_modalities="rgb",
+            proprio_obs="default",
+
+            # Unique to ManipulationRobot
+            grasping_mode="assisted",
+            
+            # unique to Behaviorbot
+            use_ghost_hands=True,
+
+            **kwargs
+    ):
+        """
+        Initializes Behaviorbot
+        Args:
+            use_ghost_hands (bool): whether to show ghost hand when the robot hand is too far away from the controller
+        """
+
+        super(Behaviorbot, self).__init__(
+            prim_path=prim_path,
+            name=name,
+            class_id=class_id,
+            uuid=uuid,
+            scale=scale,
+            visible=visible,
+            fixed_base=fixed_base,
+            visual_only=visual_only,
+            self_collisions=self_collisions,
+            load_config=load_config,
+            abilities=abilities,
+            control_freq=control_freq,
+            controller_config=controller_config,
+            action_type=action_type,
+            action_normalize=action_normalize,
+            reset_joint_pos=reset_joint_pos,
+            obs_modalities=obs_modalities,
+            proprio_obs=proprio_obs,
+            grasping_mode=grasping_mode,
+            grasping_direction="upper",
+            **kwargs,
+        )
+
+        # Basic parameters
+        self.use_ghost_hands = use_ghost_hands
+        self._world_base_fixed_joint_prim = None
+
+        # Activation parameters
+        self.first_frame = True
+        # whether hand or body is in contact with other objects (we need this since checking contact list is costly)
+        self.part_is_in_contact = {hand_name: False for hand_name in self.arm_names + ["body"]}
+
+        # Whether the VR system is actively hooked up to the VR agent.
+        self.vr_attached = False
+        self._vr_attachment_button_press_timestamp = None
```

Comment:
many of the member variables are a mixture of _var_name and var_name, maybe we should unify? I think Josiah prefers the underscore.